src/app/store/networks/networks.hooks.ts

```diff
@@ -0,0 +1,79 @@
+import { useEffect, useMemo } from 'react';
+import { useAtom } from 'jotai';
+import { StacksMainnet, StacksNetwork, StacksTestnet } from '@stacks/network';
+import { ChainID } from '@stacks/transactions';
+
+import { findMatchingNetworkKey } from '@app/common/utils';
+import { whenChainId } from '@app/common/transactions/transaction-utils';
+import { useInitialRouteSearchParams } from '@app/store/common/initial-route-search-params.hooks';
+import { useAppDispatch } from '@app/store';
+
+import { useCurrentNetwork, useNetworks } from './networks.selectors';
+import { DefaultNetworkIds, Network } from './networks.slice';
+import { networksActions } from './networks.actions';
+import { currentNetworkAtom } from './networks';
+
+// TODO: Remove - this is temporary for network refactor
+export function useSetCurrentNetworkAtom() {
+  const [_, setCurrentNetworkAtom] = useAtom(currentNetworkAtom);
+  const network = useCurrentNetwork();
+
+  useEffect(() => {
+    if (network) void setCurrentNetworkAtom(network);
+  }, [network, setCurrentNetworkAtom]);
+}
+
+export function useCurrentNetworkState() {
+  const currentNetwork = useCurrentNetwork();
+
+  return useMemo(() => {
+    const isTestnet = currentNetwork?.chainId === ChainID.Testnet;
+    const mode = isTestnet ? DefaultNetworkIds.testnet : DefaultNetworkIds.mainnet;
+    return { ...currentNetwork, isTestnet, mode };
+  }, [currentNetwork]);
+}
+
+export function useCurrentStacksNetworkState(): StacksNetwork {
+  const currentNetwork = useCurrentNetwork();
+
+  return useMemo(() => {
+    if (!currentNetwork) throw new Error('No current network');
+
+    const stacksNetwork = whenChainId(currentNetwork.chainId || 1)({
+      [ChainID.Mainnet]: new StacksMainnet({ url: currentNetwork.url }),
+      [ChainID.Testnet]: new StacksTestnet({ url: currentNetwork.url }),
+    });
+
+    stacksNetwork.bnsLookupUrl = currentNetwork.url || '';
+    return stacksNetwork;
+  }, [currentNetwork]);
+}
+
+export function useNetworksActions() {
+  const dispatch = useAppDispatch();
+
+  return useMemo(
+    () => ({
+      addNetwork({ chainId, id, name, url }: Network) {
+        dispatch(networksActions.addNetwork({ chainId, id, name, url }));
+        dispatch(networksActions.changeNetwork(id));
+        return;
+      },
+      changeNetwork(id: string) {
+        return dispatch(networksActions.changeNetwork(id));
+      },
+      removeNetwork(id: string) {
+        return dispatch(networksActions.removeNetwork(id));
+      },
+    }),
+    [dispatch]
+  );
+}
+
+export function useRequestNetworkId() {
+  const networks = useNetworks();
+  const params = useInitialRouteSearchParams();
+  const coreApiUrl = params.get('coreApiUrl');
+  const networkChainId = params.get('networkChainId');
+  return findMatchingNetworkKey({ coreApiUrl, networkChainId, networks });
```

Comment:
Does this return new reference on every render?

Reply:
Let me check, I can put it in useMemo.